modules/data-table/main/index.coffee

```diff
@@ -236,7 +236,9 @@ class DataTable extends hx.EventEmitter
       if @singleSelection() and hx.isArray(value) and value.length
         value = [value[0]]
       @_.selectedRows = new hx.Set(value)
-      @emit('selectedrowschange', {value: @_.selectedRows.values(), cause: 'api'})
+      newSelectedRows = @_.selectedRows.values()
+      @emit('selectedrowschange', {value: newSelectedRows, cause: 'api'})
+      @_.lastSelected = newSelectedRows[newSelectedRows.length - 1]
```

Comment:
this is the salient line. It sets @_.lastSelected when the selected rows were changed by the API